lib/ddtrace/auto_instrument.rb

```diff
@@ -0,0 +1,46 @@
+require 'ddtrace'
+
+module Datadog
+  # AutoInstrumentation enables all integrations
+  module AutoInstrument
+    EXCLUDED_RAILS_INTEGRATIONS = [:rack, :action_cable, :active_support, :action_pack, :action_view, :active_record].freeze
+    EXCLUDED_TEST_INTEGRATIONS = [:cucumber, :rspec].freeze
```

Comment:
If we add a new Rails integration or new test integration, and forget to add them here, I don't think any tests would fail today.

I don't have a strong preference on what approach to take, but I think we should ensure that, when a new integration is added, either by us or community, there should be no surprises with `AutoInstrument` with the integration addition.

I can see this being done in different ways:
* Having an automatic way to skip Rails sub-integrations and skip test integrations.
* Having integrations declare that they are allowed to be automatically activated (probably through a new field in `Datadog::Contrib::Patchable::ClassMethods` or similar).
* Having a whitelist of instrumentations allowed to be automatically enabled. This one is a bit worse, as it required the knowledge of a separate file than the `contrib` files added in an integration PR, but still a valid solution.

Like I said, I have no strong preference here, but I think we should address this concern.

Reply:
yea, makes sense. I tend to think `Having integrations declare that they are allowed to be automatically activated (probably through a new field in Datadog::Contrib::Patchable::ClassMethods or similar).` is the right approach here, as the `rails + test libraries` non-auto-instrumentation logic is pretty arbitrary, and i'd rather this just be declared on a per integration basis explicitly. i'll add this

Reply:
I added a `auto_instrument?` method to `Patchable::InstanceMethods` that accomplishes this. It's a bit tricky to define the logic for the rails sub-modules (or those auto-enabled by rails), since they behave differently depending on whether rails/railtie is present. But i've added some helpers here to get this to work and think it' much cleaner now

Reply:
one tradeoff here is that this PR is now quite a bit bigger but mostly it's adding boilerplate/repetitive methods to integration / integration_spec files. for your own sanity:

- looking at the `auto_instrument?` related changes to `integration/integration_spec.rb` files for a "rails-y" integration (like, active_record) is the same as all other "rails-y" integrations,  
- `cucumber / rspec` are changes identical,
- and then all the other integrations inherit the behavior and have the same base test added to their integration_spec.rb